### PR TITLE
Allow using observable IDs in `observableFormula` and `noiseFormula`

### PR DESCRIPTION
**Pull request by dweindl**
Following up on #543 and the discussion during the last PEtab editor meeting: There was general consent to allow using observableIDs in the `noiseFormula` column in the observables table.

What was not discussed explicitly:
* Should all observable IDs be allowed, or only the current one?
* If so, should it be allowed to use observable IDs inside `observableFormula`?

CC @PEtab-dev/petab-editors

Closes https://github.com/PEtab-dev/PEtab/issues/543

### CODE REVIEW

**Review by fbergmann — APPROVED** [review]
Those changes look good to me. Thanks!

**Comment by fbergmann** [comment]
As to

> What was not discussed explicitly:
> 
> * Should all observable IDs be allowed, or only the current one?

I'm fine with allowing all. 

> * If so, should it be allowed to use observable IDs inside `observableFormula`?

I don't see a problem with that, as long as this introduces no circularity, which you addressed in the PR.

**Review by dilpath — APPROVED** [review]
Looks good to me.

Could change the example to be
```
noiseParameter1_observable_pErk + noiseParameter2_observable_pErk*observable_pErk
```
instead of
```
noiseParameter1_observable_pErk + noiseParameter2_observable_pErk*pErk
```
---

> Should all observable IDs be allowed, or only the current one?

Fine for me to allow all observable IDs.
> If so, should it be allowed to use observable IDs inside `observableFormula`?

Fine for me to allow all observable IDs inside `observableFormula`.

**Review by FFroehlich — APPROVED** [review]

File: doc/documentation_data_format.rst [hunk 1]
```diff
@@ -362,7 +364,11 @@ Detailed field description
...
   Alternatively, some formula expression can be provided to specify
-  more complex noise models. A noise model which accounts for relative and
+  more complex noise models. The formula may reference any uniquely identifiable
+  model entity with PEtab-compatible identifier or any observable ID
+  specified in the observable table.
+
+  A noise model which accounts for relative and
   absolute contributions could, e.g., be defined as::
 
     noiseParameter1_observable_pErk + noiseParameter2_observable_pErk*pErk
```
Comment (dweindl) [inline doc/documentation_data_format.rst:374]:
```suggestion
    noiseParameter1_observable_pErk + noiseParameter2_observable_pErk * observable_pErk
```

**Pull request merged**